File: 14/LongestCommonPrefix.js
var longestCommonPrefix = function(strs) {
    if(!strs.length) return ""; // Edge case: empty list

    //----------------------->>> Step 1 <<<-----------------------\\ Sort array

    strs.sort();

    //----------------------->>> Step 2 <<<------------------------\\ Compare first and last word / most Different

    let first = strs[0];
    let last = strs[strs.length - 1];
    let i = 0;

    //----------------------->>> Step 3 <<<-------------------------\\ Find Common Prefix between first and last word

    while (i < first.length && first[i] === last[i]) {
        i++;
    }

    return first.substring(0, i);
}

File: Easy/14/Longest CommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";

        Array.sort(strs);

        String first = strs[0], last = strs[strs.length - 1];
        int i = 0;

        while (i < first.length() && first.charAt(i) == last.charAt(i)) {
            i++;
        }

        return first.substring(0, i);

    }
}